### DISCUSSION TITLE
Linting The Different Sources

### DISCUSSION THREAD

**Discussion by elderling** [cc65/cc65#1576] [Q&A]
I have aspirations of one day becoming a contributor to the cc65 project. Does the cc65 project itself have a standard [linter](https://en.wikipedia.org/wiki/Lint_(software)) for the C source? If no, has it considered adopting one? Does a [linter](https://en.wikipedia.org/wiki/Lint_(software)) for ca65 assembly even exist?

**Comment by mrdudz** [comment]
No such thing :)

For assembly, i doubt its needed or useful - the assembler will find all problems, and what isnt an obvious error... is almost certainly valid code :)

**Reply by elderling** [reply]
What about the C? I've used astyle to consistently format automatically, at least...

**Comment by mrdudz** [comment]
I found automatically formatting code always a bit questionable - in every non trivial project there are cases where deviating from the commonly used style is the proper thing to do :)

That said, i am more than willing to accept a PR that adds a description of the common style to the documentation - that is very much missing (for both C and asm code)
